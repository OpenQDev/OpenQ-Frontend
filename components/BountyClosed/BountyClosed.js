// Third party
import React, { useContext } from 'react';
import Link from 'next/link';

import StoreContext from '../../store/Store/StoreContext';
import useGetTokenValues from '../../hooks/useGetTokenValues';

const BountyClosed = ({ bounty, showTweetLink }) => {

	const [appState] = useContext(StoreContext);
	const [tokenValues] = useGetTokenValues(bounty?.bountyTokenBalances || []);
	const TVL = tokenValues != null && tokenValues != {}
		? appState.utils.formatter.format(tokenValues.total)
		: appState.utils.formatter.format(0);
	// Hooks
	const tweetText = `ðŸ’¸ Just claimed a developer bounty from ${bounty.owner} on OpenQ for ${TVL} working on this issue: `;
	const url = `${process.env.NEXT_PUBLIC_BLOCK_EXPLORER_BASE_URL}/tx/${bounty.claimedTransactionHash}`;
	//Render
	return (
		<div className="w-2/3 lg:w-1/2">
			<div className="flex flex-col space-y-5">
				<h2 className="flex text-3xl font-semibold  justify-center pt-16">Bounty Closed</h2>
				<div className="bg-claimed-bounty-inside col-span-3 border border-claimed-bounty rounded-lg  p-4">
					<div className='flex justify-between w-full'>
						<div>Closing Transaction: <Link href={url}>
							<span className="cursor-pointer break-all">
								<span className="underline">{url}</span>
								{'  '}<svg className="h-3 inline" fill="white" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z" />
								</svg>
							</span>
						</Link>

						</div>
						{showTweetLink && <Link
							href={`https://twitter.com/intent/tweet/?text=${tweetText}${process.env.NEXT_PUBLIC_BASE_URL}/bounty/${bounty.bountyId}/${bounty.bountyAddress}`}

						>
							<a className="hover:scale-105 animate-single-bounce duration-100 mr-2" target="_blank">
								<div id={'bounty-link'} className="cursor-pointer flex justify-end">
									<svg viewBox="0 0 128 128" width="24" height="24">
										<path d="M40.254 127.637c48.305 0 74.719-48.957 74.719-91.403 0-1.39 0-2.777-.075-4.156 5.141-4.547 9.579-10.18 13.102-16.633-4.79 2.602-9.871 4.305-15.078 5.063 5.48-4.02 9.582-10.336 11.539-17.774-5.156 3.743-10.797 6.38-16.68 7.801-8.136-10.586-21.07-13.18-31.547-6.32-10.472 6.86-15.882 21.46-13.199 35.617C41.922 38.539 22.246 26.336 8.915 6.27 1.933 20.94 5.487 39.723 17.022 49.16c-4.148-.172-8.207-1.555-11.832-4.031v.41c0 15.273 8.786 28.438 21.02 31.492a21.596 21.596 0 01-11.863.543c3.437 13.094 13.297 22.07 24.535 22.328-9.305 8.918-20.793 13.75-32.617 13.72-2.094 0-4.188-.15-6.266-.446 12.008 9.433 25.98 14.441 40.254 14.422" fill="#ffffff">
										</path>
									</svg>
								</div>
							</a>
						</Link>}
					</div>
				</div>
			</div>
		</div>
	);
};

export default BountyClosed;
