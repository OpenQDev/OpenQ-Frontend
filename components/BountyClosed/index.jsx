// Third party
import React from 'react';
import Link from 'next/link';
import Image from 'next/image';

import Twitter from '../svg/twitter';

const BountyClosed = ({ bounty, showTweetLink }) => {
  const { tvc } = bounty;
  // Hooks
  const tweetText = `ðŸ’¸ Just claimed a developer bounty from ${bounty.owner} on OpenQ for ${tvc} working on this issue: `;
  const url = `${process.env.NEXT_PUBLIC_BLOCK_EXPLORER_BASE_URL}/tx/${bounty.claimedTransactionHash}`;

  //Render
  return (
    <div className='flex px-2 sm:px-0 flex-wrap flex-1  max-w-[1200px] pb-8'>
      <div className='flex-1 pr-4 min-w-[260px] space-y-4 pt-2'>
        <h2 className='flex text-3xl'>This contract is closed.</h2>
        <h3 className='flex text-2xl border-b border-gray-700 pb-4'>
          You cannot initiate actions on a closed contract.
        </h3>
        <div className='flex border-b border-web-gray py-3 gap-2'>
          <div className='font-semibold text-muted'>Linked Closing Transaction</div>
          <div className='flex gap-1 text-primary'>
            <Link href={url} target={'_blank'} rel='noopener norefferer' className='flex items-center gap-1 underline'>
              <>
                <div id={'bounty-link'} className='flex cursor-pointer items-center'>
                  <Image src='/BountyMaterial/polyscan-white.png' width={18} height={18} alt='link-icon' />
                </div>
                <span className='underline pl-1'>view here</span>
              </>
            </Link>
          </div>
        </div>

        {showTweetLink && (
          <div className='flex items-center border-b border-web-gray py-3 gap-4'>
            <div className='font-semibold text-muted'>Tweet about it</div>
            <Link
              href={`https://twitter.com/intent/tweet/?text=${tweetText}${process.env.NEXT_PUBLIC_BASE_URL}/contract/${bounty.bountyId}/${bounty.bountyAddress}`}
              className='hover:scale-105 animate-single-bounce duration-100'
              target='_blank'
              rel='noopener noreferrer'
            >
              <>
                <Twitter />
              </>
            </Link>
          </div>
        )}
      </div>
    </div>
  );
};

export default BountyClosed;
