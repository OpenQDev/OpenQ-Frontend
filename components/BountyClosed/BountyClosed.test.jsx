/**
 * @vi-environment jsdom
 */
import React from 'react';

import BountyClosed from '.';
import Constants from '../../test-utils/constant';
import renderer from 'react-test-renderer';

describe('BountyClosed', () => {
  const bounties = Constants.bounties;

  const test = (bounty) => {
    it('should match DOM Snapshot', () => {
      const tree = renderer.create(<BountyClosed bounty={bounty} showTweetLink={true} />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  };
  bounties.forEach((elem) => {
    test(elem);
  });
});
