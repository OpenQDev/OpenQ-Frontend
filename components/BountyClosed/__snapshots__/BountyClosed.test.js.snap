// vi Snapshot v1, https://goo.gl/fbAQLP

exports[`BountyClosed should match DOM Snapshot 1`] = `
<div
  className="flex px-2 sm:px-0 flex-wrap flex-1  max-w-[1200px] pb-8"
>
  <div
    className="flex-1 pr-4 min-w-[260px] space-y-4 pt-2"
  >
    <h2
      className="flex text-3xl"
    >
      This contract is closed.
    </h2>
    <h3
      className="flex text-2xl border-b border-gray-700 pb-4"
    >
      You cannot initiate actions on a closed contract.
    </h3>
    <div
      className="flex border-b border-web-gray py-3 gap-2"
    >
      <div
        className="font-semibold text-muted"
      >
        Linked Closing Transaction
      </div>
      <div
        className="flex gap-1 text-primary"
      >
        <a
          className="flex items-center gap-1 underline"
          href="undefined/tx/undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
          rel="noopener norefferer"
          target="_blank"
        >
          <div
            className="flex cursor-pointer items-center"
            id="bounty-link"
          >
            <img
              alt="link-icon"
              data-nimg="1"
              decoding="async"
              height={18}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2FBountyMaterial%2Fpolyscan-white.png&w=48&q=75"
              srcSet="/_next/image?url=%2FBountyMaterial%2Fpolyscan-white.png&w=32&q=75 1x, /_next/image?url=%2FBountyMaterial%2Fpolyscan-white.png&w=48&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={18}
            />
          </div>
          <span
            className="underline pl-1"
          >
            view here
          </span>
        </a>
      </div>
    </div>
    <div
      className="flex items-center border-b border-web-gray py-3 gap-4"
    >
      <div
        className="font-semibold text-muted"
      >
        Tweet about it
      </div>
      <a
        className="hover:scale-105 animate-single-bounce duration-100"
        href="https://twitter.com/intent/tweet/?text=ðŸ’¸ Just claimed a developer bounty from OpenQDev on OpenQ for undefined working on this issue: undefined/contract/I_kwDOGWnnz85I9Agl/0x066efd87027a4cc4a21c57467f224ef98241d774"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          height="24"
          viewBox="0 0 128 128"
          width="24"
        >
          <path
            className="fill-primary"
            d="M40.254 127.637c48.305 0 74.719-48.957 74.719-91.403 0-1.39 0-2.777-.075-4.156 5.141-4.547 9.579-10.18 13.102-16.633-4.79 2.602-9.871 4.305-15.078 5.063 5.48-4.02 9.582-10.336 11.539-17.774-5.156 3.743-10.797 6.38-16.68 7.801-8.136-10.586-21.07-13.18-31.547-6.32-10.472 6.86-15.882 21.46-13.199 35.617C41.922 38.539 22.246 26.336 8.915 6.27 1.933 20.94 5.487 39.723 17.022 49.16c-4.148-.172-8.207-1.555-11.832-4.031v.41c0 15.273 8.786 28.438 21.02 31.492a21.596 21.596 0 01-11.863.543c3.437 13.094 13.297 22.07 24.535 22.328-9.305 8.918-20.793 13.75-32.617 13.72-2.094 0-4.188-.15-6.266-.446 12.008 9.433 25.98 14.441 40.254 14.422"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`BountyClosed should match DOM Snapshot 2`] = `
<div
  className="flex px-2 sm:px-0 flex-wrap flex-1  max-w-[1200px] pb-8"
>
  <div
    className="flex-1 pr-4 min-w-[260px] space-y-4 pt-2"
  >
    <h2
      className="flex text-3xl"
    >
      This contract is closed.
    </h2>
    <h3
      className="flex text-2xl border-b border-gray-700 pb-4"
    >
      You cannot initiate actions on a closed contract.
    </h3>
    <div
      className="flex border-b border-web-gray py-3 gap-2"
    >
      <div
        className="font-semibold text-muted"
      >
        Linked Closing Transaction
      </div>
      <div
        className="flex gap-1 text-primary"
      >
        <a
          className="flex items-center gap-1 underline"
          href="undefined/tx/undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
          rel="noopener norefferer"
          target="_blank"
        >
          <div
            className="flex cursor-pointer items-center"
            id="bounty-link"
          >
            <img
              alt="link-icon"
              data-nimg="1"
              decoding="async"
              height={18}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2FBountyMaterial%2Fpolyscan-white.png&w=48&q=75"
              srcSet="/_next/image?url=%2FBountyMaterial%2Fpolyscan-white.png&w=32&q=75 1x, /_next/image?url=%2FBountyMaterial%2Fpolyscan-white.png&w=48&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={18}
            />
          </div>
          <span
            className="underline pl-1"
          >
            view here
          </span>
        </a>
      </div>
    </div>
    <div
      className="flex items-center border-b border-web-gray py-3 gap-4"
    >
      <div
        className="font-semibold text-muted"
      >
        Tweet about it
      </div>
      <a
        className="hover:scale-105 animate-single-bounce duration-100"
        href="https://twitter.com/intent/tweet/?text=ðŸ’¸ Just claimed a developer bounty from OpenQDev on OpenQ for undefined working on this issue: undefined/contract/I_kwDOGWnnz85I9Agl/0x066efd87027a4cc4a21c57467f224ef98241d774"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          height="24"
          viewBox="0 0 128 128"
          width="24"
        >
          <path
            className="fill-primary"
            d="M40.254 127.637c48.305 0 74.719-48.957 74.719-91.403 0-1.39 0-2.777-.075-4.156 5.141-4.547 9.579-10.18 13.102-16.633-4.79 2.602-9.871 4.305-15.078 5.063 5.48-4.02 9.582-10.336 11.539-17.774-5.156 3.743-10.797 6.38-16.68 7.801-8.136-10.586-21.07-13.18-31.547-6.32-10.472 6.86-15.882 21.46-13.199 35.617C41.922 38.539 22.246 26.336 8.915 6.27 1.933 20.94 5.487 39.723 17.022 49.16c-4.148-.172-8.207-1.555-11.832-4.031v.41c0 15.273 8.786 28.438 21.02 31.492a21.596 21.596 0 01-11.863.543c3.437 13.094 13.297 22.07 24.535 22.328-9.305 8.918-20.793 13.75-32.617 13.72-2.094 0-4.188-.15-6.266-.446 12.008 9.433 25.98 14.441 40.254 14.422"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`BountyClosed should match DOM Snapshot 3`] = `
<div
  className="flex px-2 sm:px-0 flex-wrap flex-1  max-w-[1200px] pb-8"
>
  <div
    className="flex-1 pr-4 min-w-[260px] space-y-4 pt-2"
  >
    <h2
      className="flex text-3xl"
    >
      This contract is closed.
    </h2>
    <h3
      className="flex text-2xl border-b border-gray-700 pb-4"
    >
      You cannot initiate actions on a closed contract.
    </h3>
    <div
      className="flex border-b border-web-gray py-3 gap-2"
    >
      <div
        className="font-semibold text-muted"
      >
        Linked Closing Transaction
      </div>
      <div
        className="flex gap-1 text-primary"
      >
        <a
          className="flex items-center gap-1 underline"
          href="undefined/tx/undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
          rel="noopener norefferer"
          target="_blank"
        >
          <div
            className="flex cursor-pointer items-center"
            id="bounty-link"
          >
            <img
              alt="link-icon"
              data-nimg="1"
              decoding="async"
              height={18}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2FBountyMaterial%2Fpolyscan-white.png&w=48&q=75"
              srcSet="/_next/image?url=%2FBountyMaterial%2Fpolyscan-white.png&w=32&q=75 1x, /_next/image?url=%2FBountyMaterial%2Fpolyscan-white.png&w=48&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={18}
            />
          </div>
          <span
            className="underline pl-1"
          >
            view here
          </span>
        </a>
      </div>
    </div>
    <div
      className="flex items-center border-b border-web-gray py-3 gap-4"
    >
      <div
        className="font-semibold text-muted"
      >
        Tweet about it
      </div>
      <a
        className="hover:scale-105 animate-single-bounce duration-100"
        href="https://twitter.com/intent/tweet/?text=ðŸ’¸ Just claimed a developer bounty from OpenQDev on OpenQ for undefined working on this issue: undefined/contract/I_kwDOGWnnz85I9Agl/0x066efd87027a4cc4a21c57467f224ef98241d774"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          height="24"
          viewBox="0 0 128 128"
          width="24"
        >
          <path
            className="fill-primary"
            d="M40.254 127.637c48.305 0 74.719-48.957 74.719-91.403 0-1.39 0-2.777-.075-4.156 5.141-4.547 9.579-10.18 13.102-16.633-4.79 2.602-9.871 4.305-15.078 5.063 5.48-4.02 9.582-10.336 11.539-17.774-5.156 3.743-10.797 6.38-16.68 7.801-8.136-10.586-21.07-13.18-31.547-6.32-10.472 6.86-15.882 21.46-13.199 35.617C41.922 38.539 22.246 26.336 8.915 6.27 1.933 20.94 5.487 39.723 17.022 49.16c-4.148-.172-8.207-1.555-11.832-4.031v.41c0 15.273 8.786 28.438 21.02 31.492a21.596 21.596 0 01-11.863.543c3.437 13.094 13.297 22.07 24.535 22.328-9.305 8.918-20.793 13.75-32.617 13.72-2.094 0-4.188-.15-6.266-.446 12.008 9.433 25.98 14.441 40.254 14.422"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;

exports[`BountyClosed should match DOM Snapshot 4`] = `
<div
  className="flex px-2 sm:px-0 flex-wrap flex-1  max-w-[1200px] pb-8"
>
  <div
    className="flex-1 pr-4 min-w-[260px] space-y-4 pt-2"
  >
    <h2
      className="flex text-3xl"
    >
      This contract is closed.
    </h2>
    <h3
      className="flex text-2xl border-b border-gray-700 pb-4"
    >
      You cannot initiate actions on a closed contract.
    </h3>
    <div
      className="flex border-b border-web-gray py-3 gap-2"
    >
      <div
        className="font-semibold text-muted"
      >
        Linked Closing Transaction
      </div>
      <div
        className="flex gap-1 text-primary"
      >
        <a
          className="flex items-center gap-1 underline"
          href="undefined/tx/undefined"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
          rel="noopener norefferer"
          target="_blank"
        >
          <div
            className="flex cursor-pointer items-center"
            id="bounty-link"
          >
            <img
              alt="link-icon"
              data-nimg="1"
              decoding="async"
              height={18}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2FBountyMaterial%2Fpolyscan-white.png&w=48&q=75"
              srcSet="/_next/image?url=%2FBountyMaterial%2Fpolyscan-white.png&w=32&q=75 1x, /_next/image?url=%2FBountyMaterial%2Fpolyscan-white.png&w=48&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={18}
            />
          </div>
          <span
            className="underline pl-1"
          >
            view here
          </span>
        </a>
      </div>
    </div>
    <div
      className="flex items-center border-b border-web-gray py-3 gap-4"
    >
      <div
        className="font-semibold text-muted"
      >
        Tweet about it
      </div>
      <a
        className="hover:scale-105 animate-single-bounce duration-100"
        href="https://twitter.com/intent/tweet/?text=ðŸ’¸ Just claimed a developer bounty from OpenQDev on OpenQ for undefined working on this issue: undefined/contract/I_kwDOGWnnz85I9Agl/0x066efd87027a4cc4a21c57467f224ef98241d774"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          height="24"
          viewBox="0 0 128 128"
          width="24"
        >
          <path
            className="fill-primary"
            d="M40.254 127.637c48.305 0 74.719-48.957 74.719-91.403 0-1.39 0-2.777-.075-4.156 5.141-4.547 9.579-10.18 13.102-16.633-4.79 2.602-9.871 4.305-15.078 5.063 5.48-4.02 9.582-10.336 11.539-17.774-5.156 3.743-10.797 6.38-16.68 7.801-8.136-10.586-21.07-13.18-31.547-6.32-10.472 6.86-15.882 21.46-13.199 35.617C41.922 38.539 22.246 26.336 8.915 6.27 1.933 20.94 5.487 39.723 17.022 49.16c-4.148-.172-8.207-1.555-11.832-4.031v.41c0 15.273 8.786 28.438 21.02 31.492a21.596 21.596 0 01-11.863.543c3.437 13.094 13.297 22.07 24.535 22.328-9.305 8.918-20.793 13.75-32.617 13.72-2.094 0-4.188-.15-6.266-.446 12.008 9.433 25.98 14.441 40.254 14.422"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
